### ISSUE TITLE
aws_storagegateway_nfs_file_share does not populate the path attribute

### ISSUE THREAD

**Issue by acdha** [terraform-providers/terraform-provider-aws#12527]
# Description

[`aws_storagegateway_nfs_file_share` is documented](https://github.com/terraform-providers/terraform-provider-aws/blob/master/website/docs/r/storagegateway_nfs_file_share.html.markdown) as exposing the path attribute needed to mount the share:

> * `path` - File share path used by the NFS client to identify the mount point.

Unfortunately, this was not actually implemented for resource creation or updates so it cannot be used in other places within Terraform. The read path [does implemented it](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_storagegateway_nfs_file_share.go) but that appears never to be called.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.24
+ provider.aws v2.54.0
```

### Affected Resource(s)

* aws_storagegateway_nfs_file_share

### Expected Behaviour

The path attribute is available for other resources to use.

### Actual Behaviour

```
This object has no argument, nested block, or exported attribute named "path".
```

**Comment by acdha** [comment]
@bflad I took a quick pass at a patch in #12530. Not enormously familiar with the Go toolchain so feel free to ignore it if that's easier.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.55.0 of the Terraform AWS Provider, likely later today. Thanks to @acdha for the implementation. 👍

**Comment by acdha** [comment]
Thanks for the rapid response – I like the level of effort you've put into that process!
